timemachine/cpp/src/bound_potential.cu

```diff
@@ -40,6 +40,22 @@ void BoundPotential::execute_host(
     d_u.copy_to(h_u);
 };
 
+void BoundPotential::set_params_device(
+    const std::vector<int> device_shape, const double *d_new_params, const cudaStream_t stream) {
+    int updated_size = 1;
+    for (auto s : device_shape) {
+        updated_size *= s;
+    }
+    if (updated_size > 0) {
+        if (updated_size > this->size()) {
+            d_p.reset(new DeviceBuffer<double>(updated_size));
+        }
+        gpuErrchk(cudaMemcpyAsync(
+            d_p->data, d_new_params, updated_size * sizeof(*d_p->data), cudaMemcpyDeviceToDevice, stream));
+    }
+    shape = device_shape;
+}
```

Comment:
Idea for potential simplification: does BoundPotential need to know the shape of the parameters, or could it get away with just the size? It seems like switching to size would simplify a few places (and allow removing `BoundPotential::size()`)

Reply:
I agree, the shape is confusing and never used as a shape.

Am inclined to doing this in a follow up PR, to avoid expanding the scope of this PR.